packages/state-transition/src/block/processWithdrawals.ts

```diff
@@ -7,24 +7,25 @@ import {
   MAX_PENDING_PARTIALS_PER_WITHDRAWALS_SWEEP,
   FAR_FUTURE_EPOCH,
   MIN_ACTIVATION_BALANCE,
+  MAX_EFFECTIVE_BALANCE,
 } from "@lodestar/params";
 
 import {toRootHex} from "@lodestar/utils";
 import {CachedBeaconStateCapella, CachedBeaconStateElectra} from "../types.js";
 import {
   decreaseBalance,
   getValidatorMaxEffectiveBalance,
+  hasEth1WithdrawalCredential,
+  hasExecutionWithdrawalCredential,
   isCapellaPayloadHeader,
-  isFullyWithdrawableValidator,
-  isPartiallyWithdrawableValidator,
 } from "../util/index.js";
 
 export function processWithdrawals(
   fork: ForkSeq,
   state: CachedBeaconStateCapella | CachedBeaconStateElectra,
   payload: capella.FullOrBlindedExecutionPayload
 ): void {
-  const {withdrawals: expectedWithdrawals, partialWithdrawalsCount} = getExpectedWithdrawals(fork, state);
+  const {withdrawals: expectedWithdrawals, executionWithdrawalsCount} = getExpectedWithdrawals(fork, state);
```

Comment:
> refactor partialWithdrawalsCount to eip7002WithdrawalsCount to make it more meaningful. partialWithdrawalsCount sounds link to the capella withdrawal

On second thought, I am not sure about this rename. Partial withdrawals is a term introduced in Electra and also matches [naming used in the spec](https://github.com/ethereum/consensus-specs/blob/ca04b1e94de25e224308640877ef96b98ba9b119/specs/electra/beacon-chain.md?plain=1#L1022-L1023)
```python
def process_withdrawals(state: BeaconState, payload: ExecutionPayload) -> None:
    expected_withdrawals, partial_withdrawals_count = get_expected_withdrawals(state)  # [Modified in Electra:EIP7251]
```

Reply:
I see, the one from spec is also confusing because we already have full withdrawals and partial withdrawals since capella
how about `electraPartialWithdrawalsCount`?

Reply:
But in capella, there was no difference in handling full vs. partial withdrawal, they were both considered as a withdrawal just with a different amount.

I don't see us using the term "partial withdrawal" on unstable branch, so I am wondering if we even need the electra prefix to differentiate in the code

Reply:
I reverted the change, added comments instead

Reply:
> Partial withdrawals is a term introduced in Electra and also matches [naming used in the spec](https://github.com/ethereum/consensus-specs/blob/ca04b1e94de25e224308640877ef96b98ba9b119/specs/electra/beacon-chain.md?plain=1#L1022-L1023)

I believe the term is introduced in Capella. Back then the definition is straightforward:
Partial withdrawal: A withdrawal with amount < 32eth that's generated from withdrawal sweep
Full withdrawal: A withdrawal with amount equals to entire validator's balance generated from withdrawal sweep (underlying reason could be being slashed, voluntary exit etc.)

In Electra, we have EL triggered partial withdrawal aka withdrawal request aka execution withdrawal aka execution request withdrawal. The associated field in beacon state is `pending_partial_withdrawals`. 

EIP-7002's partial withdrawal is enhanced to support full withdrawal in maxEB when the amount is set to 0. But the associate spec (eg. `state.pending_partial_withdrawals`, `partial_withdrawals_count` in `get_expected_withdrawal`) are not updated. 

So @twoeths's motive to rename this is reasonable. Something like `execution_withdrawals_count` sound pretty reasonable to me. However we should change the spec (and it's a good idea) first before updating the related naming in Lodestar though.

Reply:
> Something like execution_withdrawals_count sound pretty reasonable to me

yeah that was my first suggestion (https://github.com/ChainSafe/lodestar/pull/7045#discussion_r1729845058) and initial thought but after reading up on spec I wasn't sure it's a good rename.

> However we should change the spec (and it's a good idea) first before updating the related naming in Lodestar though.

sounds reasonable to suggest this rename on the spec as well

Reply:
> EIP-7002's partial withdrawal is enhanced to support full withdrawal in maxEB when the amount is set to 0. But the associate spec (eg. state.pending_partial_withdrawals, partial_withdrawals_count in get_expected_withdrawal) are not updated.

This statement is wrong. Execution requests with 0 amount are filtered during `process_withdrawal_request` and will not be appended to `state.pending_partial_withdrawals` so the naming in the spec is fine (including `partial_withdrawals_count` in `get_expected_withdrawals`. 

Still need to update the spec to make it clearer to differentiate between partial withdrawals from partial withdrawal sweep and partial withdrawals from execution payload though